src/ErrorPage.php

```diff
@@ -0,0 +1,339 @@
+		$this->exception = $exception;
+		return $this;
+	}
+
+	/**
+	 * Set the response_code of the error page.
+	 *
+	 * @param string $response_code Response code to use.
+	 * @return self
+	 */
+	public function with_response_code( $response_code ) {
+		$this->response_code = $response_code;
+		return $this;
+	}
+
+	/**
+	 * Render the error page.
+	 *
+	 * This first sets the required headers and then returns the HTML to send as
+	 * output.
+	 *
+	 * @return string
+	 */
+	public function render() {
+		$this->send_to_error_log();
+
+		$this->set_headers();
+
+		$text_direction = function_exists( 'is_rtl' ) && is_rtl()
+			? 'rtl'
+			: 'ltr';
+
+		$styles = $this->get_styles( $text_direction );
+		$html   = $this->get_html( $styles, $text_direction );
+
+		return $html;
+	}
+
+	/**
+	 * Send the exception that was caught to the error log.
+	 */
+	private function send_to_error_log() {
+		error_log( // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+			sprintf(
+				"%s - %s (%s) [%s]\n%s",
+				$this->message,
+				$this->exception->getMessage(),
+				$this->exception->getCode(),
+				get_class( $this->exception ),
+				$this->exception->getTraceAsString()
+			)
+		);
+	}
+
+	/**
+	 * Sets the required headers.
+	 *
+	 * This will only adapt the headers if they haven't yet been sent.
+	 */
+	private function set_headers() {
+		if ( ! headers_sent() ) {
+			// Define the content type for the error page.
+			header( 'Content-Type: text/html; charset=utf-8' );
+
+			// Mark the page as a server failure so it won't get indexed.
+			status_header( $this->response_code );
+
+			// Let the browser know this result shouldn't get cached.
+			nocache_headers();
+		}
+	}
+
+	/**
+	 * Get the HTML output for the error page.
+	 *
+	 * @param string $styles         CSS styles to use.
+	 * @param string $text_direction Text direction. Can be 'ltr' or 'rtl'.
+	 * @return string HTML output.
+	 */
+	private function get_html( $styles, $text_direction ) {
+		$no_robots = function_exists( 'wp_no_robots' )
+			? wp_no_robots()
+			: '';
+
+		return <<<HTML
+<!DOCTYPE html>
+<html dir="{$text_direction}">
+	<head>
+		<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+		<meta name="viewport" content="width=device-width">
+		{$no_robots}
+		<title>{$this->title}</title>
+		{$styles}
+	</head>
+	<body id="error-page">
+		<h1>{$this->title}</h1>
+		<p>{$this->message}</p>
+		{$this->render_exception()}
+	</body>
+</html>
+HTML;
+	}
+
+	/**
+	 * Render the exception of the error page.
+	 *
+	 * The exception details are only rendered if both WP_DEBUG and
+	 * WP_DEBUG_DISPLAY are true.
+	 *
+	 * @return string HTML describing the exception that was thrown.
+	 */
+	private function render_exception() {
+		if ( null === $this->exception ) {
+			return '';
+		}
+
+		if ( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG ) {
+			return '';
+		}
+
+		if ( ! defined( 'WP_DEBUG_DISPLAY' ) || ! WP_DEBUG_DISPLAY ) {
+			return '';
```

Comment:
Perhaps this method should advise to enable WP_DEBUG_DISPLAY to see the error, or else look at the PHP error log.